x-pack/platform/plugins/shared/fleet/common/constants/agentless.ts

```diff
@@ -15,6 +15,25 @@ export const AGENTLESS_GLOBAL_TAG_NAME_ORGANIZATION = 'organization';
 export const AGENTLESS_GLOBAL_TAG_NAME_DIVISION = 'division';
 export const AGENTLESS_GLOBAL_TAG_NAME_TEAM = 'team';
 
+export const MAXIMUM_RETRIES = 3;
+
+const HTTP_500_INTERNAL_SERVER_ERROR = 500;
+const HTTP_501_NOT_IMPLEMENTED = 501;
+const HTTP_502_BAD_GATEWAY = 502;
+const HTTP_503_SERVICE_UNAVAILABLE = 503;
+const HTTP_504_GATEWAY_TIMEOUT = 504;
+
+const ECONNREFUSED_CODE = 'ECONNREFUSED';
+
+export const RETRYABLE_HTTP_STATUSES = [
```

Comment:
Not completely clear what server these are targetting, but ES can also return 429 status which means "we're too busy! try again later" - which means it's retryable, but you should wait ...

Reply:
@pmuellr thanks for the review! `RETRYABLE_HTTP_STATUSES` are for Agentless API which is hosted in MKI environment. We had logging error instance  where the Agentless API return `500` logging error 
[Agentless API] Upgrading the agentless agent failed with a status 500 internal service error -`{"code":"BACKEND_OPERATION_FAILED","error":"failed to update agentless config: Operation cannot be fulfilled on xxx": the object has been modified; please apply your changes to the latest version and try again"}} {"url":"xxx","method":"PUT","data":{"stack_version":"xxx","fleet_token":"[REDACTED]"},...`

Reply:
Just as a note, we fixed this error in the API (or at least we think so)

Reply:
Oleg suggested adding retries so unless guaranteed, I rather keep the retry just in case.